### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by totakke**
This PR includes the following updates.

- clojure 1.11
- clojure libraries
- actions
  - java 17
  - codecov v3
  - cljs tests with node

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -28,31 +28,29 @@ jobs:
     - name: Run tests
       run: lein with-profile +${{ matrix.clojure }} test
     - name: Run cljs tests
-      run: lein with-profile +${{ matrix.clojure }} doo phantom test once
+      run: lein with-profile +${{ matrix.clojure }} doo node test once
```
Comment (totakke) [inline .github/workflows/build.yml:31]: PhantomJS development was suspended. https://phantomjs.org/

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -28,31 +28,29 @@ jobs:
...
+        java-version: '17'
     - name: Install dependencies
       run: lein deps
     - name: Generate code coverage
-      run: |
-        lein cloverage --codecov
-        bash <(curl -s https://codecov.io/bash) -f target/coverage/codecov.json
-    - uses: codecov/codecov-action@v1
+      run: lein cloverage --codecov
+    - uses: codecov/codecov-action@v3
```
Comment (totakke) [inline .github/workflows/build.yml:53]: codecov-action v1 (bash uploader) was [deprecated](https://github.com/codecov/codecov-action#%EF%B8%8F--deprecation-of-v1).

File: src/clj_hgvs/internal.cljc [hunk 3]
```diff
@@ -1,5 +1,6 @@
 (ns clj-hgvs.internal
   "Common functions used only inside this project."
+  (:refer-clojure :exclude [parse-long])
```
Comment (totakke) [inline src/clj_hgvs/internal.cljc:3]: `clojure.core/parse-long` was added in clojure 1.11.

**Review by federkasten — APPROVED** [review]
LGTM :ok_man:

**Pull request merged**